### DISCUSSION TITLE
Plugins aware

### DISCUSSION THREAD

**Discussion by lgbm** [ant-media/Ant-Media-Server#6095] [Plugins/Custom Extensions]
Is there a way to listen publisher and subscriber streams start/stop? Something like IStreamAwareScopeHandler (streamSubscriberStart streamSubscriberStop, etc). 
Given others usefull things like subscribers list, deny publish and play, etc.

Regards,
Luis

**Comment by USAMAWIZARD** [comment]
Hi Luis,
you can listen for [stream started](https://github.com/ant-media/Plugins/blob/7af1c402289a07497cc54c62897735324b2a3049/SamplePlugin/src/main/java/io/antmedia/plugin/SamplePlugin.java#L72) and [stream Finished ](https://github.com/ant-media/Plugins/blob/7af1c402289a07497cc54c62897735324b2a3049/SamplePlugin/src/main/java/io/antmedia/plugin/SamplePlugin.java#L77C14-L77C28) in the plugin , but there is no direct callback available for streamSubscriberStart or streamSubscriberStop

Can you please give me some more details about what you are trying to archive , Thank you.
